src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -426,6 +472,16 @@ static std::vector<Range> getHyperrectangleForRowGroup(const parquet::FileMetaDa
     return hyperrectangle;
 }
 
+static std::vector<ParquetBloomFilterCondition::ConditionElement> createConditionsFromRpn(parquet::BloomFilterReader & bf_reader,
+                                                                                          const Block & header,
+                                                                                          const std::vector<std::pair<std::string, int>> & column_name_to_index,
+                                                                                          const std::shared_ptr<const KeyCondition> & key_condition,
+                                                                                          const std::unordered_set<std::string> & filtering_columns)
+{
+    const auto column_index_to_bf = buildColumnIndexToBF(bf_reader, 0, header, column_name_to_index, filtering_columns);
```

Comment:
BF data is not needed for building the condition. AFAICT, it's used in 2 ways:
 1. Checking which columns have bf. But this information is available in `parquet::FileMetaData` which we already have in memory: `metadata.RowGroup(row_group_idx)->ColumnChunk(0)->bloom_filter_offset().has_value()` . Can pass the `const parquet::RowGroupMetaData &` to `keyConditionRPNToParquetBloomFilterCondition` (don't forget the `if (!header.has(column_name))`).
 2. Constructing a boilerplate hasher object. But it's just always `parquet::XxHasher`.

This will allow constructing `filtering_columns` based on `ParquetBloomFilterCondition` instead of `KeyCondition`, which seems important as it'll exclude common non-bf conditions like `key > 42`.